### PR TITLE
[IMP] payment_alipay: conversion to payment apocalypse 

### PR DESCRIPTION
**Pull request by arj-odoo**
Before this commit, payment_alipay used the old payment API.

Taskid: 2333027

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment_alipay/controllers/main.py [hunk 1]
```diff
@@ -14,54 +14,33 @@
...
-            elif resp == 'TRADE_CLOSED':
-                _logger.warning('Alipay: payment refunded to user and closed the transaction')
-            else:
-                _logger.warning('Alipay: unrecognized alipay answer, received %s instead of TRADE_FINISHED/TRADE_SUCCESS and TRADE_CLOSED' % (post['trade_status']))
-        if post.get('out_trade_no') and post.get('trade_no'):
-            # TODO ARJ wtf? do we really need to do this?
-            post['reference'] = request.env['payment.transaction'].sudo().search([('reference', '=', post['out_trade_no'])]).reference
-            return request.env['payment.transaction'].sudo()._handle_feedback_data('alipay', post)
-        return False  # TODO ARJ useless rtype bool, could be None
+    _return_url = '/payment/alipay/return'  # If these are constant, they should be modified in all modules
```
Comment (arj-odoo) [inline addons/payment_alipay/controllers/main.py:17]: @AntoineVDV I think it's quite clear like that. We need to change it everywhere if we change the convention. Up to you.

File: addons/payment_alipay/controllers/main.py [hunk 2]
```diff
@@ -14,54 +14,33 @@
...
         # keep re-sending notification, until over 24 hour 22 minutes
         # Generally, there are 8 notifications within 25 hours (Frequency: 2m,10m,15m,1h,2h,6h,15h).
         if response.text == 'true':
-            self._alipay_validate_data(**post)
+            request.env['payment.transaction'].sudo()._handle_feedback_data('alipay', post)
             return 'success'
-        return ""  # TODO ARJ nitpicking: returning "fail" would still be a 200 but would clearly state that the notif couldn't be processed
+        return "fail"
 
     # TODO ARJ there is still the not so likely eventuality of an HTTP 307 but I think that we can safely drop the POST
```
Comment (arj-odoo) [inline addons/payment_alipay/controllers/main.py:38]: ? drop the parameters? They are used to validate the transaction.
Maybe you mean only keep the GET ?

File: addons/payment_alipay/controllers/main.py (hunk 1 above)
Comment (AntoineVDV) [reply addons/payment_alipay/controllers/main.py:17]: Watch out, you're on the odoo/odoo PR. I'll close it to keep all discussions on the other repo.

About my comment: As it said, it was really nitpicking so there is no need to change it. Nonetheless, I don't believe that there is a convention for constants. Some are located in a `const.py` file, some are upper-case variables in the scope of the file, while here (and in other places) they are private class attributes.

File: addons/payment_alipay/controllers/main.py (hunk 2 above)
Comment (AntoineVDV) [reply addons/payment_alipay/controllers/main.py:38]: Yes

**Pull request closed without merging**